lib/menu.rb

```diff
@@ -0,0 +1,13 @@
+class Menu
+  attr_accessor :dish_list
```

Comment:
This could be made an `attr_reader`. Why might we prefer that?

Reply:
I used an attr_accessor because my idea was to be able to create a new dish and push it to the menu dish list, but maybe it is better to use an attr_reader and include a method in the takeaway class, something like "update menu", that receives a dish as a parameter and then adds it, in that way the attribute can't be modified from outside.

Reply:
Absolutely correct — when designing class interfaces, we want to lock down what the user of the class can do as much as possible. The more control you have over what they can do with your state, the fewer possible situations the class can be in, the fewer bugs are possible in your code, and the fewer unit tests you need.

For example: in this case, the user of your class (who might not be you) might get the format of `:dish_list` wrong or set it to nil. By locking it down with an attr_reader and/or an `add_dish` method, you remove that possibility.